build_script.py

```diff
@@ -184,29 +257,34 @@ def main(args=sys.argv[1:]):
     delegates building or testing XCTest to a sub-parser and its corresponding
     function.
     """
+    strategy = DarwinStrategy if platform.system() == 'Darwin' else LinuxStrategy
```

Comment:
I've actually never tried building corelibs-xctest on FreeBSD, but I imagine it should succeed using the `LinuxStrategy`. However, I've been told that it's something of a faux pas to refer to FreeBSD as Linux. Could we rename this to `UnixStrategy` or `GenericUnixStrategy`? This would match the naming scheme of [Swift's toolchains](https://github.com/apple/swift/blob/2af7041af32c0fbda8d1a77a605905c0a99c8d02/lib/Driver/ToolChains.h#L39).